### ISSUE TITLE
libreadline is not declared as a dependency in meson build

### ISSUE THREAD

**Issue by twaik** [albfan/miraclecast#346]
```
[14/39] Compiling C object 'src/ctl/e16b67d@@miracle-wifictl@exe/ctl-cli.c.o'.
FAILED: src/ctl/e16b67d@@miracle-wifictl@exe/ctl-cli.c.o 
cc -Isrc/ctl/e16b67d@@miracle-wifictl@exe -Isrc/ctl -I../src/ctl -Isrc/ctl/../.. -I../src/ctl/../.. -Isrc/shared -I../src/shared -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu11 -O2 -g -D_GNU_SOURCE -DBUILD_ENABLE_DEBUG  -MD -MQ 'src/ctl/e16b67d@@miracle-wifictl@exe/ctl-cli.c.o' -MF 'src/ctl/e16b67d@@miracle-wifictl@exe/ctl-cli.c.o.d' -o 'src/ctl/e16b67d@@miracle-wifictl@exe/ctl-cli.c.o' -c ../src/ctl/ctl-cli.c
../src/ctl/ctl-cli.c:37:10: fatal error: readline/history.h: No such file or directory
 #include <readline/history.h>
          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
[19/39] Compiling C object 'src/dhcp/0d1ba6f@@miracle-dhcp@exe/client.c.o'.
ninja: build stopped: subcommand failed.
```

**Comment by albfan** [comment]
it is

https://github.com/albfan/miraclecast/blob/master/meson.build#L23

so the problem is about how the define wraps this code

**Comment by albfan** [comment]
Thanks for pointing this out, there's no way to make it optional (in autotools it wasn't)

Fixed on 4111333
